components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/throttling/ThrottleHandler.java

```diff
@@ -229,127 +230,131 @@ private boolean doRoleBasedAccessThrottlingWithCEP(MessageContext synCtx, Config
-
-                        // Checking Applicability of Conditions is a relatively expensive operation. So we are
-                        // going to check it only if the API/Resource is throttled out.
-                        if (getThrottleDataHolder().isAPIThrottled
-                                (resourceLevelThrottleKey)) {
-                            if (getThrottleDataHolder().isConditionsAvailable(resourceLevelThrottleKey)) {
-                                Map<String, List<ConditionDto>> conditionDtoMap = getThrottleDataHolder()
-                                        .getConditionDtoMap(resourceLevelThrottleKey);
-                                if (log.isDebugEnabled()) {
-                                    log.debug("conditions available" + conditionDtoMap.size());
-                                }
-                                String throttledCondition = getThrottleConditionEvaluator().getThrottledInCondition
-                                        (synCtx, authContext, conditionDtoMap);
-                                if (StringUtils.isNotEmpty(throttledCondition)) {
+                    //If verbInfo is present then only we will do resource level throttling
+                    if (isUnlimittedTier && !apiLevelThrottledTriggered) {
+                        //If unlimited tier throttling will not apply at resource level and pass it
+                        if (log.isDebugEnabled()) {
+                            log.debug("Resource level throttling set as unlimited and request will pass " +
+                                    "resource level");
+                        }
+                    } else {
+                        if (APIConstants.API_POLICY_USER_LEVEL.equalsIgnoreCase(verbInfoDTO.getApplicableLevel())) {
+                            resourceLevelThrottleKey = resourceLevelThrottleKey + "_" + authorizedUser;
+                            policyLevelUserTriggered = true;
+                        }
+                        //If tier is not unlimited only throttling will apply.
+                        List<String> resourceLevelThrottleConditions = new ArrayList<>();
+                        resourceLevelThrottleConditions = verbInfoDTO.getThrottlingConditions();
+                        conditionGroupDTOs = verbInfoDTO.getConditionGroups();
+
+                        Timer timer1 = getTimer(MetricManager.name(
+                                APIConstants.METRICS_PREFIX, this.getClass().getSimpleName(), RESOURCE_THROTTLE));
+                        Timer.Context
+                                context1 = timer1.start();
+
+                        if (conditionGroupDTOs != null && conditionGroupDTOs.length > 0) {
+                            // Checking Applicability of Conditions is a relatively expensive operation. So we are
+                            // going to check it only if the API/Resource is throttled out.
+                            if (getThrottleDataHolder().isAPIThrottled
```

Comment:
does this line if combined with the line below exceed 120 characters?

Reply:
Yes.